Illiar/feat/kyb signup

chesterkmr opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

- **New Features**
	- Introduced a new `SignUpPage` component with a dynamic signup form.
	- Added error handling components for missing and invalid access tokens.
	- Implemented a new `GlobalProviders` component to manage global context providers.
	- Added a streamlined routing structure with nested routes for improved navigation.
	- Introduced a `Background`, `Header`, `Footer`, and `Content` components for the signup layout.
	- Added a `SignUpForm` component to facilitate user sign-up functionality.
	- Introduced new error handling for various routing errors.
	- Enhanced the `CustomerProvider` to navigate users to the signup page if required.
	- Added a new `ErrorScreen` component for handling different routing errors.
	- Introduced a `Submit` component for the signup form.
	- Added a `DependenciesProvider` for managing dependencies and loading states.

- **Bug Fixes**
	- Enhanced error handling in the `CustomerProvider` to check for signup requirements.

- **Chores**
	- Updated dependencies in `package.json` to include `sonner` for notifications.

- **Documentation**
	- Updated component exports for better organization and accessibility.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tomer-shvadron committed:
feat: signup page backend

chesterkmr committed:
feat: reorganized routes & added initial signup layout & reworked providers

tomer-shvadron committed:
feat: small changes to enable token creation without enduser

chesterkmr committed:
feat: updated signup layout & added signup markup

tomer-shvadron committed:
feat: pr comments fixes

tomer-shvadron committed:
Merge branch 'dev' into bal-3039

tomer-shvadron committed:
feat: added integration test for collection-flow.signup.controller

tomer-shvadron committed:
feat: remove console.log

tomer-shvadron committed:
feat: small cr comment fix

chesterkmr committed:
Merge branch 'bal-3039' into illiar/feat/kyb_signup

chesterkmr committed:
feat: finalized signup

chesterkmr committed:
fix: removed unused code

chesterkmr committed:
fix: cleanup

chesterkmr committed:
feat: updated endpoints & fixed theme & loader fixes

chesterkmr committed:
fix: cleanup

chesterkmr committed:
fix: refactor

chesterkmr committed:
feat: signup form schema update

chesterkmr committed:
Merge branch 'dev' into illiar/feat/kyb_signup

tomer-shvadron commented on apps/kyb-app/src/common/providers/AccessTokenProvider/AccessTokenProvider.tsx:
```diff
@@ -0,0 +1,50 @@
+    [accessToken, setAccessToken],
+  );
+
+  const applyAccessTokenToUrlIfNeeded = useCallback(
```
You can combine lines 26-47 into one simple `useEffect`

tomer-shvadron commented on apps/kyb-app/src/pages/Root/Root.tsx:
```diff
@@ -0,0 +1,28 @@
+  const { accessToken } = useAccessToken();
+
+  useEffect(() => {
+    if (isLoading) return;
```
why inside the useEffect and not outside?

chesterkmr replied:
If you talking about redirect then this condition won't ever render outlet and there will be blank screen.

tomer-shvadron replied:
Usually we have the `isLoading` checks not inside `useEffect`, can you just check if it can be pulled out?

tomer-shvadron commented on apps/kyb-app/src/pages/Root/Root.tsx:
```diff
@@ -0,0 +1,28 @@
+  }, [isSignupRequired, isLoading]);
+
+  useEffect(() => {
+    if (isLoading) return;
```
Same

chesterkmr replied:
Same

chesterkmr committed:
fix: refactor

tomer-shvadron committed:
feat: renamed collection-flow/signup controller

chesterkmr committed:
fix: refactor

tomer-shvadron committed:
refactor(collection-flow): rename signup controller to no-user controller

- Update all references from CollectionFlowSignupController to CollectionFlowNoUserController
- Adjust API endpoints to reflect the new no-user functionality

(Your code has changed so much, it’s like a butterfly emerging from a very confused caterpillar)

chesterkmr committed:
Merge branch 'dev' into illiar/feat/kyb_signup

tomer-shvadron committed:
feat: bad migration fix

tomer-shvadron committed:
feat: added bad migration fix

tomer-shvadron committed:
feat: quick fix for db issue

tomer-shvadron committed:
feat: revert db cleanup fix

chesterkmr committed:
fix: added missing main representative on cf signup

chesterkmr committed:
fix: fixed tests & mainRepresentative
